src/utils/docker.ts

```diff
@@ -18,7 +18,8 @@ export function projectId(cid: string): string {
 }
 
 export function getServicePort(queryEndpoint: string): number | undefined {
-  return queryEndpoint ? Number(queryEndpoint.split(':')[2]) : undefined;
+  const port = queryEndpoint ? queryEndpoint.split(':')[2] : undefined;
+  return port ? Number(port) : undefined;
```

Comment:
when port is not a number, `Number(port)` will return NA which won't be treated as falsy value by `? : ` operator

Reply:
Maybe you should change the default value from undefined to 3100

Reply:
We don't need to return the default port, the reason to keep it working just not breaking the port.service which has dependency on it. It has no impact on the runtime service